--- origsrc/gedit-2.28.0/configure.ac	2009-09-21 09:58:06.000000000 -0500
+++ src/gedit-2.28.0/configure.ac	2009-10-29 22:38:26.569199900 -0500
@@ -60,8 +60,6 @@ AC_MSG_CHECKING([for Win32 platform])
 case "$host" in
   *-*-mingw*|*-*-cygwin*)
     platform_win32=yes
-    AC_CHECK_TOOL(WINDRES, windres)
-    ACTIVE_PLUGINS="${ACTIVE_PLUGINS},checkupdate"
     ;;
   *)
     platform_win32=no
@@ -74,6 +72,8 @@ AC_MSG_CHECKING([for native Win32])
 case "$host" in
   *-*-mingw*)
     os_win32=yes
+    AC_CHECK_TOOL(WINDRES, windres)
+    ACTIVE_PLUGINS="${ACTIVE_PLUGINS},checkupdate"
     ;;
   *)
     os_win32=no
@@ -418,8 +418,12 @@ fi
 # so we create a lib
 if test "$platform_win32" = yes; then
 	GEDIT_IMPLIB="-lgedit-$GEDIT_API_VERSION"
-	PLUGIN_LIBTOOL_FLAGS="-module -avoid-version -no-undefined -mwindows -Wl,\$(top_builddir)/gedit/libgedit-\$(GEDIT_API_VERSION).a"
-	LOADER_LIBTOOL_FLAGS="-module -avoid-version -no-undefined -mwindows -Wl,\$(top_builddir)/gedit/libgedit-\$(GEDIT_API_VERSION).a"
+	PLUGIN_LIBTOOL_FLAGS="-module -avoid-version -no-undefined -Wl,\$(top_builddir)/gedit/libgedit-\$(GEDIT_API_VERSION).a"
+	LOADER_LIBTOOL_FLAGS="-module -avoid-version -no-undefined -Wl,\$(top_builddir)/gedit/libgedit-\$(GEDIT_API_VERSION).a"
+	if test "$os_win32" = yes; then
+		PLUGIN_LIBTOOL_FLAGS="${PLUGIN_LIBTOOL_FLAGS} -mwindows"
+		LOADER_LIBTOOL_FLAGS="${LOADER_LIBTOOL_FLAGS} -mwindows"
+	fi
 else
 	GEDIT_IMPLIB=""
 	PLUGIN_LIBTOOL_FLAGS="-module -avoid-version"
--- origsrc/gedit-2.28.0/gedit/Makefile.am	2009-09-20 12:43:50.000000000 -0500
+++ src/gedit-2.28.0/gedit/Makefile.am	2009-10-29 22:38:26.579199900 -0500
@@ -21,7 +21,10 @@ gedit_SOURCES = \
 gedit_LDADD = libgedit.la $(GEDIT_LIBS) $(EGG_SMCLIENT_LIBS)
  
 if PLATFORM_WIN32
-gedit_LDFLAGS = -Wl,--export-all-symbols -mwindows -Wl,--out-implib,libgedit-$(GEDIT_API_VERSION).a
+gedit_LDFLAGS = -Wl,--export-all-symbols -Wl,--out-implib,libgedit-$(GEDIT_API_VERSION).a
+if OS_WIN32
+gedit_LDFLAGS += -mwindows
+endif
 else
 gedit_LDFLAGS = -export-dynamic -no-undefined -export-symbols-regex "^[[^_]].*"
 endif
@@ -45,7 +48,7 @@ BUILT_SOURCES = 			\
 	gedit-marshal.c			\
 	gedit-marshal.h
 
-if PLATFORM_WIN32
+if OS_WIN32
 gedit-res.o: gedit.rc
 	$(WINDRES) -i gedit.rc --input-format=rc -o gedit-res.o -O coff
 
@@ -111,7 +114,7 @@ headerdir = $(prefix)/include/gedit-@GED
 header_DATA = 				\
 	$(INST_H_FILES)	
 
-if !PLATFORM_WIN32
+if !OS_WIN32
 POSIXIO_FILES = 			\
 	gedit-local-document-saver.c	
 else
@@ -217,7 +220,7 @@ else
 	ln -s gedit $(DESTDIR)$(bindir)/gnome-text-editor
 endif
 
-if !PLATFORM_WIN32
+if !OS_WIN32
 BACON_DIR=$(srcdir)/../../libbacon/src/
 BACON_FILES=bacon-message-connection.h bacon-message-connection.c
 
