--- origsrc/gedit-2.30.3/gedit/Makefile.am	2010-06-20 14:43:02.000000000 -0500
+++ src/gedit-2.30.3/gedit/Makefile.am	2010-06-28 15:30:10.947717900 -0500
@@ -226,9 +226,10 @@ install-exec-hook:
 if PLATFORM_WIN32
 	$(mkinstalldirs) "$(DESTDIR)$(libdir)"
 	$(INSTALL_DATA) libgedit-$(GEDIT_API_VERSION).a "$(DESTDIR)$(libdir)"
-else
+endif
+if !OS_WIN32
 	rm -f $(DESTDIR)$(bindir)/gnome-text-editor
-	ln -s gedit $(DESTDIR)$(bindir)/gnome-text-editor
+	ln -s gedit$(EXEEXT) $(DESTDIR)$(bindir)/gnome-text-editor
 endif
 
 if !OS_WIN32
