--- origsrc/gedit-3.0.0/configure.ac	2011-04-04 11:12:51.000000000 -0500
+++ src/gedit-3.0.0/configure.ac	2011-04-11 11:59:47.097507000 -0500
@@ -412,8 +412,8 @@ fi
 # on win32 plugins must be able to resolve all symbols at link time
 # so we create a lib
 if test "$platform_win32" = yes; then
-	GEDIT_IMPLIB="-lgedit-$GEDIT_API_VERSION"
-	PLUGIN_LIBTOOL_FLAGS="-module -avoid-version -no-undefined -Wl,\$(top_builddir)/gedit/libgedit.a"
+	GEDIT_IMPLIB="-lgedit"
+	PLUGIN_LIBTOOL_FLAGS="-module -avoid-version -no-undefined -Wl,\$(top_builddir)/gedit/libgedit.a"
 	if test "$os_win32" = yes; then
 		PLUGIN_LIBTOOL_FLAGS="${PLUGIN_LIBTOOL_FLAGS} -mwindows"
 	fi
--- origsrc/gedit-3.0.0/gedit/Makefile.am	2011-03-27 08:14:54.000000000 -0500
+++ src/gedit-3.0.0/gedit/Makefile.am	2011-04-11 12:02:29.353389900 -0500
@@ -235,7 +235,7 @@ if !ENABLE_GVFS_METADATA
 libgedit_la_SOURCES += gedit-metadata-manager.c
 endif
 
-if !PLATFORM_WIN32
+if !OS_WIN32
 libgedit_la_SOURCES += gedit-fifo.c
 NOINST_H_FILES += gedit-fifo.h
 endif
@@ -281,11 +281,11 @@ if HAVE_INTROSPECTION
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS = Gedit-3.0.gir
 
-Gedit-3.0.gir: gedit
+Gedit-3.0.gir: gedit$(EXEEXT)
 INTROSPECTION_SCANNER_ARGS = -I$(top_srcdir) --warn-all
 Gedit_3_0_gir_NAMESPACE = Gedit
 Gedit_3_0_gir_VERSION = 3.0
-Gedit_3_0_gir_PROGRAM = $(builddir)/gedit
+Gedit_3_0_gir_PROGRAM = $(builddir)/gedit$(EXEEXT)
 Gedit_3_0_gir_FILES = $(INST_H_FILES) $(libgedit_c_files) $(BUILT_SOURCES)
 Gedit_3_0_gir_INCLUDES = Gtk-3.0 GtkSource-3.0
 
@@ -307,9 +307,10 @@ install-exec-hook:
 if PLATFORM_WIN32
 	$(mkinstalldirs) "$(DESTDIR)$(libdir)"
 	$(INSTALL_DATA) libgedit.a "$(DESTDIR)$(libdir)"
-else
+endif
+if !OS_WIN32
 	rm -f $(DESTDIR)$(bindir)/gnome-text-editor
-	ln -s gedit $(DESTDIR)$(bindir)/gnome-text-editor
+	ln -s gedit$(EXEEXT) $(DESTDIR)$(bindir)/gnome-text-editor
 endif
 
 -include $(top_srcdir)/git.mk
