--- origsrc/gedit-2.22.3/configure.ac	2008-05-27 07:05:34.000000000 -0500
+++ src/gedit-2.22.3/configure.ac	2008-06-03 00:15:17.218750000 -0500
@@ -343,8 +343,22 @@
    AC_SUBST(DISABLE_DEPRECATED_CFLAGS)
 fi
 
-PLUGIN_LIBTOOL_FLAGS="-module -avoid-version"
+case "$host_os" in
+  cygwin*|mingw*)
+    platform_win32=yes
+    GEDIT_IMPLIB="-lgedit-$GEDIT_API_VERSION"
+    PLUGIN_LIBTOOL_FLAGS="-module -avoid-version -no-undefined -Wl,\$(top_builddir)/gedit/libgedit-\$(GEDIT_API_VERSION).a"
+    ;;
+  *)
+    platform_win32=no
+    GEDIT_IMPLIB=""
+    PLUGIN_LIBTOOL_FLAGS="-module -avoid-version"
+    ;;
+esac
+
+AC_SUBST(GEDIT_IMPLIB)
 AC_SUBST(PLUGIN_LIBTOOL_FLAGS)
+AM_CONDITIONAL(PLATFORM_WIN32, test "x$platform_win32" = "xyes")
 
 AC_CONFIG_FILES([
 Makefile
--- origsrc/gedit-2.22.3/data/gedit.desktop.in	2008-05-27 07:15:17.000000000 -0500
+++ src/gedit-2.22.3/data/gedit.desktop.in	2008-06-03 01:05:26.000000000 -0500
@@ -14,4 +14,4 @@
 X-GNOME-Bugzilla-Product=gedit
 X-GNOME-Bugzilla-Component=general
 X-GNOME-Bugzilla-Version=2.22.3
-X-GNOME-Bugzilla-ExtraInfoScript=/opt/gnome/libexec/gedit-2/gedit-bugreport.sh
+X-GNOME-Bugzilla-ExtraInfoScript=/usr/sbin/gedit-2/gedit-bugreport.sh
--- origsrc/gedit-2.22.3/data/gedit.pc.in	2008-05-25 05:34:53.000000000 -0500
+++ src/gedit-2.22.3/data/gedit.pc.in	2008-06-03 00:15:05.921875000 -0500
@@ -8,3 +8,4 @@
 Requires: libgnomeui-2.0 libglade-2.0 gtksourceview-2.0
 Version: @VERSION@
 Cflags: -I${includedir}/gedit-@GEDIT_API_VERSION@
+Libs: -L${libdir} @GEDIT_IMPLIB@
--- origsrc/gedit-2.22.3/gedit/Makefile.am	2008-05-25 05:34:54.000000000 -0500
+++ src/gedit-2.22.3/gedit/Makefile.am	2008-06-03 00:15:17.250000000 -0500
@@ -30,20 +30,25 @@
 gedit_SOURCES = gedit.c
 gedit_LDADD = libgedit.la $(GEDIT_LIBS)
 
+if PLATFORM_WIN32
+gedit_LDFLAGS = -Wl,--export-all-symbols -Wl,--out-implib,libgedit-$(GEDIT_API_VERSION).a
+else
 gedit_LDFLAGS = -export-dynamic -no-undefined -export-symbols-regex "^[[^_]].*"
-
+endif
 
 libgedit_la_LDFLAGS = -export-dynamic -no-undefined -export-symbols-regex "^[[^_]].*"
 
-libgedit_la_LIBADD =			\
-	$(GEDIT_LIBS)			\
-	dialogs/libdialogs.la
+libgedit_la_LIBADD =
 
 if ENABLE_PYTHON
 libgedit_la_LIBADD += \
 	$(top_builddir)/bindings/python/gedit.la
 endif
 
+libgedit_la_LIBADD +=			\
+	dialogs/libdialogs.la	\
+	$(GEDIT_LIBS)
+
 BUILT_SOURCES = 			\
 	gedit-enum-types.c		\
 	gedit-enum-types.h		\
@@ -211,7 +216,11 @@
 
 install-exec-hook:
 	rm -f $(DESTDIR)$(bindir)/gnome-text-editor
-	ln -s gedit $(DESTDIR)$(bindir)/gnome-text-editor
+	ln -s gedit$(EXEEXT) $(DESTDIR)$(bindir)/gnome-text-editor
+if PLATFORM_WIN32
+	$(mkinstalldirs) "$(DESTDIR)$(libdir)"
+	$(INSTALL_DATA) libgedit-$(GEDIT_API_VERSION).a "$(DESTDIR)$(libdir)"
+endif
 
 BACON_DIR=$(srcdir)/../../libbacon/src/
 BACON_FILES=bacon-message-connection.h bacon-message-connection.c
